snuba/datasets/replacements/rate_limiter.py

```diff
@@ -0,0 +1,66 @@
+import time
+from contextlib import contextmanager
+from enum import Enum
+from typing import Iterator, MutableMapping, Optional, Tuple
+
+from snuba import state
+
+RATE_LIMIT_PER_SEC = "mem_rate_limit_per_sec_"
+
+# Stores the rate limit current state.
+# The key is the bucket to keep multiple instances logically apart
+# The value contains the epoch and the number of attempt in that
+# epoch
+BUCKETS: MutableMapping[str, Tuple[int, int]] = {}
```

Comment:
Two things about this variable:

1. I don't like having mutable global state without at least a level of indirection. I would prefer it if this was put behind an accessor like the [SerializableExceptionRegistry](https://github.com/getsentry/snuba/blob/master/snuba/utils/serializable_exception.py#L78-L85). 
2. If this isn't thread-safe it should be made threadlocal

Reply:
I removed the entire global state, made the rate limiter module reusable (it is in snuba/utils now) and it is now threadsafe.